lib/msf/core/post/linux/compile.rb

```diff
@@ -54,7 +61,11 @@ def upload_and_compile(path, data, gcc_args='')
   end
 
   def strip_comments(c_code)
-    c_code.gsub(%r{/\*.*?\*/}m, '').gsub(%r{^\s*//.*$}, '')
+    c_code.gsub(%r{^\s*/\*[\S\s]*?\*/}, '').gsub(%r{^\s*//.*$}, '')
```

Comment:
The original code only stripped `/* abcdefg */` style comments if they were on the same line and the containing line did not start with whitespace.  In retrospect writing this explanation,, we should probably reevaluate this to allow stripping a comment like:
```
x=14; /* This is still still not stripped */
```

Reply:
Waiiiiit...... no I need to look at this again.... the original should have worked.

Reply:
Yup; it worked fine in ruby, but not on the regex parser I was using for testing....